docs: Add architecture overview (#2816)
oasislabs/oasis-core: docs/consensus/transactions.md

# Transactions

The consensus layer uses a common transaction format for all transactions. As

> everything else it tries to be independent of any concrete [consensus backend].

Reviewer: ```suggestion
with other Oasis Core components, it tries to be independent of any concrete [consensus backend].
```

* `amount` is the total fee amount to be paid.
* `gas` is the maximum gas that an operation can use.

## Gas Estimation

> As transactions need to include the maximum amount of gas that can be consumed

Reviewer: ```suggestion
As transactions need to provide the maximum amount of gas that can be consumed
```
